graphql/subscription.go

```diff
@@ -44,9 +44,13 @@ func (s *subscriptionMap) Unsubscribe(subscriptionID string) error {
 	if !success {
 		return fmt.Errorf("tried to unsubscribe from unknown subscription with ID '%s'", subscriptionID)
 	}
+	hasBeenUnsubscribed := unsub.hasBeenUnsubscribed
 	unsub.hasBeenUnsubscribed = true
 	s.map_[subscriptionID] = unsub
-	reflect.ValueOf(s.map_[subscriptionID].interfaceChan).Close()
+
+	if hasBeenUnsubscribed {
```

Comment:
Isn't this backwards?
```suggestion
	if !hasBeenUnsubscribed {
```

Maybe a test would help confirm, although I'm not sure how hard that is to set up, maybe it's not worthwhile.